service/policy/objects.proto

```diff
@@ -265,16 +265,36 @@ message KeyAccessServer {
   common.Metadata metadata = 100;
 }
 
+// A KAS public key and some associated metadata for further identifcation
+message KasPublicKey {
+  // x509 ASN.1 content in PEM envelope, usually
+  string pem = 1;
+
+  // A unique string identifier for this key
+  string kid = 2;
+
+  // A known algorithm type with any additional parameters encoded
+  string alg = 3;
+}
+
+// A list of known KAS public keys
+message KasPublicKeySet {
+  repeated KasPublicKey keys = 1;
+}
+
 message PublicKey {
   oneof public_key {
     // kas public key url - optional since can also be retrieved via public key
     string remote = 1 [(buf.validate.field).cel = {
-      id: "uri_format",
-      message: "URI must be a valid URL (e.g., 'https://demo.com/') followed by additional segments. Each segment must start and end with an alphanumeric character, can contain hyphens, alphanumeric characters, and slashes.",
+      id: "uri_format"
+      message: "URI must be a valid URL (e.g., 'https://demo.com/') followed by additional segments. Each segment must start and end with an alphanumeric character, can contain hyphens, alphanumeric characters, and slashes."
       expression: "this.matches('^https://[a-zA-Z0-9]([a-zA-Z0-9\\\\-]{0,61}[a-zA-Z0-9])?(\\\\.[a-zA-Z0-9]([a-zA-Z0-9\\\\-]{0,61}[a-zA-Z0-9])?)*(/.*)?$')"
     }];
 
     // public key - optional since can also be retrieved via url
     string local = 2;
+
+    // public key with additional information. Current preferred version
+    KasPublicKeySet cached = 3;
```

Comment:
This construct should replace local, not exist alongside it, correct? A base64-encoded public key (which was the prior value for local in this `oneof`) has no use going forward once this `KasPublicKeySet cached` construct is added?

Reply:
it is a oneof so this provides backwards compatibility. I can mark it deprecated if you want

Reply:
Thanks - please mark it deprecated and reserved. 🙏  I'd prefer not to preserve backwards compatibility with an early API implementation that was never fully utilized or supported downstream.